lib/waylon/jenkins/job/rest.rb

```diff
@@ -26,13 +26,13 @@ def status
 
         def est_duration
           # estimatedDuration is returned in ms; here we convert it to seconds
-          client.api_get_request("/job/#{@name}/lastBuild", nil, '/api/json?depth=1&tree=estimatedDuration')['estimatedDuration'] / 1000
+          @est_duration ||= client.api_get_request("/job/#{URI.escape @name}/lastBuild", nil, '/api/json?depth=1&tree=estimatedDuration')['estimatedDuration'] / 1000
```

Comment:
It was already being memcached https://github.com/puppetlabs/waylon/blob/master/lib/waylon/jenkins/job/memcached.rb#L21
so we would only refresh that value every minute with current ttl of the cache.
The multiple hits to the api were seen with me testing with caching off so presumable not a big issue in production.